services/device-registry-base/src/main/java/org/eclipse/hono/service/management/credentials/CredentialsDto.java

```diff
@@ -201,25 +242,30 @@ public void createMissingSecretIds() {
     }
 
     /**
-     * Merges the secrets of the given credential DTO with that of the current one.
+     * Merges the secrets of the given credential DTO with those of this one.
      *
-     * @param otherCredentialsDto The credential DTO to be merged.
-     * @return  a reference to this for fluent use.
+     * @param otherCredentialsDto The DTO to be merged into this one.
+     * @return A reference to this for fluent use.
      * @throws NullPointerException if the given credentials DTO is {@code null}.
+     * @throws IllegalArgumentException if the given credentials DTO does not contain any credentials
      */
     @JsonIgnore
     public CredentialsDto merge(final CredentialsDto otherCredentialsDto) {
         Objects.requireNonNull(otherCredentialsDto);
 
-        Optional.ofNullable(otherCredentialsDto.getCredentials())
-                .ifPresent(credentialsToMerge -> this.getData()
-                        .forEach(credential -> findCredentialByIdAndType(credential.getAuthId(), credential.getType(),
-                                credentialsToMerge)
-                                        .ifPresent(credential::merge)));
+        final var credentialsToMerge = otherCredentialsDto.getCredentials();
+        if (credentialsToMerge == null || credentialsToMerge.isEmpty()) {
+            throw new IllegalArgumentException("DTO does not contain any credentials");
```

Comment:
With respect to issue #2735, FMPOV the issue is that we are able to update with secret-ids that do not exist already provided the existing credentials list is empty. If the existing credentials list is not empty, then currently we send the error message `other credential has no secret with id: xyz`. But with this fix, in the case of empty list of existing credentials the error message is `DTO does not contain any credentials` and IMHO it is bit misleading. Shouldn't we keep it consistent with the other one?

Reply:
What is misleading about it? We could still change it to e.g.
> no credentials on record for device

Or did you have anything specific in mind?

Reply:
FMPOV the issue is that a user is trying to update with secret-ids that do not exist. In this scenario, IMHO an error message such as  `no secrets are available for the given secret ids` or `other credential has no secret with id: xyz` (in consistent with the other scenario). Also `no credentials on record for device` suits too.

Reply:
I have rebased and updated the error message ...